### PR TITLE
fix(controller): improve domain name validation per RFC 1123

### LINKED ISSUE #2979: deis blocks domain creation under improper circumstances
`$ deis domains:add django.paas-sandbox`
`Adding django.paas-sandbox to rubber-undertow... 400 BAD REQUEST`
`{u'domain': [u'Hostname does not look like a valid hostname. Only lowercase characters are allowed.']}`

deis will not let me create a domain where the top of the domain is "paas-sandbox", as displayed above, however this is not a top level domain nor is it intended as one. Our corporation uses an internal domain that is set as the dns search on all corporate computers. For example, if someone types `http://www/` in their browser, it will come up because their dns searches `corp.example.com` for non qualifying domains. They are really going to the site `www.corp.example.com` but all they type and all the host header on the server sees is "www". deis requires the domain to exist how the user will display it in the browser in order to render the page. Our DNS system is almost identical to Active Directory so anyone using Active Directory can relate but this is not a MS / Active Directory specific scenario. This can be done using only dhcp and most ISP's will do this from their dhcp as well.

Anyways, please fix. Thank you.

### PR DESCRIPTION
**Pull request by mboersma**
Closes #2979, replaces #2983.

### CODE REVIEW

File: controller/api/serializers.py [hunk 1]
```diff
@@ -239,23 +239,22 @@ def validate_domain(self, value):
         """
         Check that the hostname is valid
         """
-        match = re.match(
-            r'^(\*\.)?(' + settings.APP_URL_REGEX + r'\.)*([a-z0-9-]+)\.([a-z0-9]{2,})$',
-            value)
-        if not match:
+        if len(value) > 255:
```
Comment (bacongobbler) [inline controller/api/serializers.py]: I believe a Validator already exists for this: https://github.com/django/django/blob/master/django/core/validators.py#L294

There's some pretty useful validators at `django.core.validators`. We should use them more often :)

Comment (bacongobbler) [inline controller/api/serializers.py]: note: we can use these validators if we shift this over to the model, which ay be a better idea (I did this in https://github.com/deis/deis/pull/2898)

File: controller/api/tests/test_domain.py [hunk 2]
```diff
@@ -48,24 +48,34 @@ def test_response_data(self):
...
-        self.assertEqual(response.status_code, 204)
-        url = '/v1/apps/{app_id}/domains'.format(app_id=self.app_id)
-        response = self.client.get(url, content_type='application/json',
-                                   HTTP_AUTHORIZATION='token {}'.format(self.token))
-        self.assertEqual(0, response.data['count'])
+        test_domains = [
+            'test-domain.example.com',
+            'django.paas-sandbox',
+            'domain',
+            'not.too.loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong',
```
Comment (bacongobbler) [inline controller/api/tests/test_domain.py]: :laughing:

**Comment by bacongobbler** [comment]
we can probably clean this up and move the validation to the model, but this works for now. Code LGTM!

**Comment by gabrtv** [comment]
Code LGTM too.  Nice work with the test coverage!

**Pull request merged**